src/subgraphs.jl

```diff
@@ -0,0 +1,110 @@
+#Given a graph g, form 'nsubgraphs' subgraphs. Try to keep all subgraphs the same size and minimise edges cut between them
+#Return the subgraphs (1... nsubgraphs) with their contained vertices. Also return a dictionary of the subgraphs connected to each sgraph
+#KaHyPar needs to be installed for this function
+function formsubgraphs(g::NamedDimGraph, nsubgraphs::Int64)
+
+    ps = partition(g, nsubgraphs, configuration = :edge_cut, imbalance = 0.0)
+
+    subgraphconns = Dict{Int, Vector{Int}}()
```

Comment:
Can `subgraphconns` just be a `Graph`?

Reply:
In fact this seems to call for using a `DataGraph`, where the vertex metadata is the `Vector{Tuple}` storing the vertices associated with a particular partition.

Reply:
That seems like a good idea and would make using the subgraphs in routines a bit more straightforward. I will try to re-write it in that way.

Reply:
I have implemented this change in the latest commit. This should make the code more streamlined.